Documentation added for clusterclass (#634)
vmware-tanzu/cluster-api-provider-bringyourownhost: docs/cluster_class.md

              infrastructureCluster: true
          jsonPatches:
            - op: add
              path: "/spec/template/spec/bundleLookupBaseRegistry"
              valueFrom:
                variable: bundleLookupBaseRegistry
    - name: controlPlaneEndpoint
      description: "Sets control plane endpoint"
      definitions:
        - selector:
            apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
            kind: ByoClusterTemplate
            matchResources:
              infrastructureCluster: true
          jsonPatches:
            - op: add
              path: "/spec/template/spec/controlPlaneEndpoint"
              valueFrom:
                template: |
                  host: '{{ .controlPlaneIpAddr }}'
                  port: 6443
    - name: kubeVipEnabled
      definitions:
        - jsonPatches:
            - op: add
              path: /spec/template/spec/kubeadmConfigSpec/files/0/content
              valueFrom:
                variable: kubeVipPodManifest
          selector:
            apiVersion: controlplane.cluster.x-k8s.io/v1beta1
            kind: KubeadmControlPlaneTemplate
            matchResources:
              controlPlane: true
```

> To generate a cluster using this clusterclass, refer below yaml where we create just one `kind` named `Cluster`

Reviewer: ```suggestion
To generate a cluster using this ClusterClass, refer below yaml where we create just one `kind` named `Cluster`
```

                value: "10"
              - name: vip_retryperiod
                value: "2"
              image: ghcr.io/kube-vip/kube-vip:v0.4.1
              imagePullPolicy: IfNotPresent
              name: kube-vip
              resources: {}
              securityContext:
                capabilities:
                  add:
                  - NET_ADMIN
                  - NET_RAW
              volumeMounts:
              - mountPath: /etc/kubernetes/admin.conf
                name: kubeconfig
            hostNetwork: true
            hostAliases:
              - hostnames:
                  - kubernetes
                ip: 127.0.0.1
            volumes:
            - hostPath:
                path: /etc/kubernetes/admin.conf
                type: FileOrCreate
              name: kubeconfig
          status: {}
    workers:
      machineDeployments:
        - class: quickstart-worker
          metadata: {}
          name: md-0
          replicas: 1
```

### How to use with clusterctl?

To create a cluster using clusterctl with managed topology, use `topology` flavor and set following environment variables :
```
CLUSTER_NAME
CONTROL_PLANE_ENDPOINT_IP
```

> Also set feature flag for clusterclass:

Reviewer: ```suggestion
Also set feature flag for ClusterClass:
```

**Feature gate name:** `ClusterTopology`

**Variable name to enable/disable the feature gate**: `CLUSTER_TOPOLOGY`

After setting the variables use below command to create a yaml file to create a cluster with managed topology
```shell
clusterctl generate cluster <cluster-name> --kubernetes-version <kubernetes-version> --control-plane-machine-count=3 --worker-machine-count=3 \
    --flavor topology > my-cluster.yaml
```

> #### clusterctl alpha topology plan

Reviewer 2: ```suggestion
### clusterctl alpha topology plan
```
Just to make it a distinct heading

Author: Actually it's not a separate heading, and it's under clusterctl, so doesn't feel to make it distinct.

To preview  the Cluster generated using such ClusterClass, use `clusterctl alpha topology plan`

```shell
clusterctl alpha topology plan -f example-cluster-class.yaml -f example-cluster.yaml -o output/
```

> To know more on clusterclass, refer [this](https://cluster-api.sigs.k8s.io/tasks/experimental-features/cluster-class/index.html) documentaion.

Reviewer: ```suggestion
To know more about ClusterClass, refer to [this](https://cluster-api.sigs.k8s.io/tasks/experimental-features/cluster-class/index.html) documentation.
```
